plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,311 @@
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+
+import static org.autorefactor.refactoring.ASTHelper.*;
+import static org.eclipse.jdt.core.dom.ASTNode.*;
+import static org.eclipse.jdt.core.dom.Assignment.Operator.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.Refactorings;
+
+/*
+ * TODO when findViewById is reusing a local variable,
+ * the viewholderitem will create a new field with duplicate name.
+ * Possible solution: use the id names instead of var names
+ * TODO sometimes convertView is a final param in getView.
+ * Possible solution: remove final modifier
+ */
+
+/** See {@link #getDescription()} method. */
+public class AndroidViewHolderRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Optimization for Android applications to optimize getView routines. "
+                + "It allows reducing the calls to inflate and getViewById Android " + "API methods.";
+    }
+
+    @Override
+    public String getName() {
+        return "Android ViewHolder";
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        IMethodBinding methodBinding = node.resolveBinding();
+        Block body = node.getBody();
+        if (body != null && isMethod(methodBinding, "android.widget.Adapter", "getView", "int", "android.view.View",
+                "android.view.ViewGroup")) {
+            GetViewVisitor visitor = new GetViewVisitor();
+            body.accept(visitor);
+            if (visitor.viewVariable != null && !visitor.isInflateInsideIf()) {
+                final ASTBuilder b = this.ctx.getASTBuilder();
+                final Refactorings r = this.ctx.getRefactorings();
+
+                // Transform tree
+
+                // Create If statement
+                // test-clause
+                InfixExpression condition = b.infixExpr(b.simpleName("convertView"), InfixExpression.Operator.EQUALS,
+                        b.null0());
+                // then
+                Assignment assignment = b.assign(b.simpleName("convertView"), Assignment.Operator.ASSIGN,
+                        b.copy(visitor.getInflateExpression()));
+                Block thenBlock = b.block(b.toStmt(assignment));
+
+                IfStatement ifStatement = b.if0(condition, thenBlock);
+                r.insertBefore(ifStatement, visitor.viewAssignmentStatement);
+
+                // assign to local view variable when necessary
+                if (!"convertView".equals(visitor.viewVariable.getIdentifier())) {
+                    Statement assignConvertViewToView = null;
+                    if (visitor.viewVariableDeclarationFragment != null) {
+                        assignConvertViewToView = b.declareStmt(
+                                b.type(visitor.viewVariable.resolveTypeBinding().getName()),
```

Comment:
It helps decide how to write a type name based on the imports already present in the class (Can we use the simple name or do we need to resort to fully qualified name?)